grep/icon_grep/util/utils.py

```diff
@@ -1,58 +1,33 @@
-import komand
+from insightconnect_plugin_runtime.exceptions import PluginException
+from subprocess import run, CalledProcessError
 import tempfile
-import shutil
 
-# Specify path for unix utils grep and wc:
-grep = "egrep"
-wc = "wc"
 
-FOUND = "found"
-HITS = "hits"
-MATCHES = "matches"
-TEMP_PATH = tempfile.mkdtemp() + "/"
+def run_grep(log: object, text: str, pattern: str, behavior: str) -> str:
+    with tempfile.NamedTemporaryFile() as fp:
+        fp.write(text.encode())
```

Comment:
Do we need to impose any limit on the file size here? What will happen if the data is more than can be stored temporarily in the plugin container?

Reply:
This something we should think about for sure.